cpp/src/shuffler/postbox.cpp

```diff
@@ -48,10 +114,10 @@ std::vector<std::tuple<PartID, ChunkID, std::size_t>> PostBox::search(MemoryType
 ) const {
     std::lock_guard const lock(mutex_);
     std::vector<std::tuple<PartID, ChunkID, std::size_t>> ret;
-    for (auto& [pid, chunks] : pigeonhole_) {
-        for (auto& [cid, chunk] : chunks) {
+    for (auto const& [pid, chunks] : pigeonhole_) {
+        for (auto const& chunk : chunks) {
             if (chunk.gpu_data && chunk.gpu_data->mem_type() == mem_type) {
-                ret.emplace_back(pid, cid, chunk.gpu_data->size);
+                ret.emplace_back(pid, chunk.cid, chunk.gpu_data->size);
```

Comment:
I think this whole changeset seems much more complicated than it needs to be.

For example `PostBox::search` is only used in `postbox_spilling`. It doesn't need to return `PartID, ChunkID` pairs, it could just as easily return `Rank, ChunkID` pairs. I think.